crates/astria-cli/src/commands/bridge.rs

```diff
@@ -0,0 +1,484 @@
+use std::{
+    path::PathBuf,
+    sync::Arc,
+    time::Duration,
+};
+
+use astria_bridge_contracts::i_astria_withdrawer::{
+    IAstriaWithdrawer,
+    Ics20WithdrawalFilter,
+    SequencerWithdrawalFilter,
+};
+use astria_core::{
+    bridge::{
+        self,
+        Ics20WithdrawalFromRollupMemo,
+    },
+    primitive::v1::{
+        asset::{
+            self,
+            TracePrefixed,
+        },
+        Address,
+    },
+    protocol::transaction::v1alpha1::{
+        action::{
+            BridgeUnlockAction,
+            Ics20Withdrawal,
+        },
+        Action,
+    },
+};
+use clap::Args;
+use color_eyre::eyre::{
+    self,
+    bail,
+    eyre,
+    OptionExt as _,
+    WrapErr as _,
+};
+use ethers::{
+    contract::EthEvent,
+    core::types::Block,
+    providers::{
+        Middleware,
+        Provider,
+        ProviderError,
+        StreamExt as _,
+        Ws,
+    },
+    types::{
+        Filter,
+        Log,
+        H256,
+    },
+};
+use tokio::select;
+use tracing::warn;
+
+#[derive(Args, Debug)]
+pub struct CollectWithdrawalEvents {
+    #[arg(long)]
+    rollup_endpoint: String,
+    #[arg(long)]
+    contract_address: ethers::types::Address,
+    #[arg(long)]
+    next_rollup_block_height: u64,
```

Comment:
it might be nice to also have a `to_block` option instead of always using the current head block by default

Reply:
I have renamed this line to `from_rollup_height` and added an arg `to_rollup_height: Option<u64>` in [f513e17](https://github.com/astriaorg/astria/pull/1261/commits/f513e1783620d4e6cdfb54ca2111935a3f1685f1).

The stream is now bounded if `--to-rollup-height` is set, or unbounded otherwise (requiring ctrl-c to exit).